crates/re_viewer/src/ui/time_panel/time_axis.rs

```diff
@@ -101,3 +43,115 @@ impl TimelineAxis {
     //     self.ranges.last().max
     // }
 }
+
+// in seconds or nanos
+fn time_abs_diff(a: TimeInt, b: TimeInt) -> u64 {
+    a.as_i64().abs_diff(b.as_i64())
+}
+
+/// First determine the threshold for when a gap should be closed.
+/// Sometimes, looking at data spanning milliseconds, a single second pause can be an eternity.
+/// When looking at data recorded over hours, a few minutes of pause may be nothing.
+/// We also don't want to produce a timeline of only gaps.
+/// Finding a perfect heuristic is impossible, but we do our best!
+fn gap_size_heuristic<T>(time_type: TimeType, times: &BTreeMap<TimeInt, T>) -> u64 {
+    crate::profile_function!();
+
+    assert!(!times.is_empty());
+
+    if times.len() <= 2 {
+        return u64::MAX;
+    }
+
+    let total_time_span = time_abs_diff(
+        *times.first_key_value().unwrap().0,
+        *times.last_key_value().unwrap().0,
+    );
+
+    // We start off by a minimum gap size - any gap smaller than this will never be collapsed.
+    // This is partially an optimization, and partially something that "feels right".
+    let min_gap_size: u64 = match time_type {
+        TimeType::Sequence => 9,
+        TimeType::Time => 100_000_000, // nanos!
```

Comment:
I sort of prefer the reader having to do their own math rather than adding a comment that is going to be outdated the next time someone changes that line